qml/Ping360EthernetConfiguration.qml

```diff
@@ -0,0 +1,118 @@
+import QtQuick 2.7
+import QtQuick.Controls 2.2
+import QtQuick.Controls.Material 2.1
+import QtQuick.Layouts 1.3
+import QtQml.Models 2.2
+
+import DeviceManager 1.0
+import Ping360HelperService 1.0
+import StyleManager 1.0
+
+PingPopup {
+    id: root
+
+    closePolicy: Popup.NoAutoClose
+    property var connection: undefined
+
+    // Default size
+    width: 300
+    height: 280
+
+    function configureConnection(connection) {
+        open()
+        root.connection = connection
+    }
+
+    Label {
+        id: title
+        text: "Ping360 Ethetnet configuration"
+        anchors.horizontalCenter: parent.horizontalCenter
+    }
+
+    ListModel {
+        id: ethernetModes
+        ListElement {
+            name: "DHCP Client"
+            info: "Allows to connect as a normal ethernet client to a DHCP server."
+        }
+        ListElement {
+            name: "Static IP"
+            info: "Used to set a fixed IP address in a personalized network environment.<br>⚠ <strong>Ensure to configure your subnet if necessary</strong> ⚠"
+        }
+    }
+
+    GridLayout {
+        id: grid
+        columns: 2
+        anchors.top: title.bottom
+        anchors.right: parent.right
+        anchors.left: parent.left
+        anchors.bottom: parent.bottom
+
+        RowLayout {
+            Layout.columnSpan: 2
+            Layout.fillWidth: true
+            Label {
+                text: "Type:"
+            }
+            PingComboBox {
+                id: comboBox
+                model: ethernetModes
+                textRole: "name"
+                Layout.fillWidth: true
+            }
+        }
+
+        Text {
+            text: ethernetModes.get(comboBox.currentIndex).info
+            color: comboBox.currentIndex == 1 ? "red" : "black"
+            horizontalAlignment: Text.AlignHCenter | Text.AlignJustify
+            wrapMode: Text.Wrap
+            Layout.columnSpan: 2
+            Layout.fillWidth: true
+        }
+
+        PingTextField {
+            id: ipTextField
+            title: "IP address:"
+            text: "192.168.2.4"
+            Layout.columnSpan: 2
+            Layout.alignment: Qt.AlignHCenter
+            visible: comboBox.currentIndex == 1
+            // TODO: Move to RegularExpressionValidator when using Qt 5.14+
+            // Check for valid IPV4 (0.0.0.0, 192.168.0.2)
+            validator: RegExpValidator {
+                regExp: /^(?!\.)((^|\.)([1-9]?\d|1\d\d|2(5[0-5]|[0-4]\d))){4}$/gm
```

Comment:
Nitpick: The annoying thing with this is that it doesn't let you erase all numbers before writing new ones. I wish it let you type but just didn't use the value instead... Do you think there is a better way?

Reply:
Can you share a gif ?  I was unable to replicate it.

Reply:
It'll not allow invalid values, if you start to type a invalid IP address, you'll stop you to typing.

Reply:
exactly, I just dislike that behavior, but I guess it is a Qt thing.

Reply:
if you have 192.168.2.4, you can't erase the 4 and then write a 5, you need to write a 5 and then erase the 4

Reply:
keep it as is, if anyone complains we can think about improving it